src/sdc11073/provider/providerimpl.py

```diff
@@ -490,7 +494,10 @@ def stop_realtime_sample_loop(self):
             self.waveform_provider.stop()
 
     def get_xaddrs(self) -> list[str]:
-        addresses = self._wsdiscovery.get_active_addresses()  # these own IP addresses are currently used by discovery
+        if self._use_hostname:
+            addresses = [socket.getfqdn()]
```

Comment:
please provide a test for your new feature

Reply:
done, new test+consumer/provider connection added.